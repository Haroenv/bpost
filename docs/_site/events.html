<!DOCTYPE html>
<head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Getting package events | Modern JavaScript API wrapper for bpost track & trace</title>
<link rel="stylesheet" href="css/syntax.css">
<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
<link rel="stylesheet" href="css/modern-business.css">
<link rel="stylesheet" href="css/lavish-bootstrap.css">
<link rel="stylesheet" href="css/customstyles.css">
<link rel="stylesheet" href="css/theme-blue.css">
<link rel="shortcut icon" href="images/favicon.ico">

</head>
<body>
  <!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="fa fa-home fa-lg navbar-brand" href="index.html">&nbsp;<span class="projectTitle"> Bpost</span></a>
        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                
                
                
                <li><a href="https://github.com/thibmaek/bpost/releases" target="_blank">Changelog</a></li>
                
                
                
                <li><a href="https://github.com/thibmaek/bpost/issues" target="_blank">Issues</a></li>
                
                
                
                <li><a href="https://github.com/thibmaek/bpost/projects" target="_blank">Roadmap</a></li>
                
                
                
                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                
                
                <!--comment out this block if you want to hide search-->
                <li>
                    <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                            SimpleJekyllSearch.init({
                                searchInput: document.getElementById('search-input'),
                                resultsContainer: document.getElementById('results-container'),
                                dataSource: 'search.json',
                                searchResultTemplate: '<li><a href="{url}" title="Getting package events">{title}</a></li>',
                    noResultsText: 'No results found.',
                            limit: 10,
                            fuzzy: true,
                    })
                    </script>
                    <!--end search-->
                </li>
            </ul>
        </div>
        </div>
        <!-- /.container -->
</nav>

  <main class="container">
    <div class="col-lg-12">&nbsp;</div>
    <div class="row">
      <div class="col-md-3">
      

  <ul id="mysidebar" class="nav">
    <li class="sidebarTitle">bpost 1.0.0</li>
    
      
      <li>
        <a href="#">Getting Started</a>
        <ul>
          
            
              <li><a href="installing.html">Installing</a></li>
            
          
            
              <li><a href="constructing.html">Constructing a session</a></li>
            
          
        </ul>
      </li>
      
      <li>
        <a href="#">Methods</a>
        <ul>
          
            
              <li><a href="status.html">Getting active status</a></li>
            
          
            
              <li><a href="packageinfo.html">Getting package Info</a></li>
            
          
            
              <li class="active"><a href="events">Getting package events</a></li>
            
          
            
              <li><a href="homedelivery.html">Home delivery options</a></li>
            
          
            
              <li><a href="postalpoint.html">Postal point options</a></li>
            
          
        </ul>
      </li>
      
    
  </ul>
</div>

        <div class="col-md-9">
        <div class="post-header">
   <h1 class="post-title-main">Getting package events</h1>
</div>

<div class="post-content">
  

  
  <a target="_blank" rel="noopener" href="https://github.com/thibmaek/bpost/tree/master/docs/pages/events.md" class="btn btn-default githubEditButton" role="button"><i class="fa fa-github fa-lg"></i> Edit me</a>
  

  <h2 id="droppedoffbysender"><code class="highlighter-rouge">#droppedOffBySender()</code></h2>

<h3 id="description">Description</h3>
<p><code class="highlighter-rouge">bpost.droppedOffBySender()</code> will return information when the package is dropped off by the sender at senderâ€™s source postal point.
It requires one passed parameter <code class="highlighter-rouge">id</code> and returns the data with the <code class="highlighter-rouge">key: 'received.domestic-corner',</code> value in the API response.</p>

<h3 id="spec">Spec</h3>

<table>
  <thead>
    <tr>
      <th>Parameters</th>
      <th>Return type</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code class="highlighter-rouge">(id: String)</code></td>
      <td><code class="highlighter-rouge">Promise&lt;Object&gt;</code></td>
    </tr>
  </tbody>
</table>

<h3 id="usage">Usage</h3>
<div class="language-js highlighter-rouge"><pre class="highlight"><code><span class="kr">import</span> <span class="p">{</span> <span class="nx">droppedOffBySender</span> <span class="p">}</span> <span class="nx">from</span> <span class="s1">'bpost'</span><span class="p">;</span>

<span class="nx">droppedOffBySender</span><span class="p">(</span><span class="err">`</span><span class="mi">323267009691651472</span><span class="err">`</span><span class="p">).</span><span class="nx">then</span><span class="p">(</span><span class="nx">res</span> <span class="o">=&gt;</span> <span class="nx">res</span><span class="p">);</span>
</code></pre>
</div>

<h3 id="output">Output</h3>
<div class="language-js highlighter-rouge"><pre class="highlight"><code><span class="p">{</span>
  <span class="s2">"date"</span><span class="err">:</span> <span class="s2">"2016-11-09"</span><span class="p">,</span>
  <span class="s2">"irregularity"</span><span class="err">:</span> <span class="kc">false</span><span class="p">,</span>
  <span class="s2">"key"</span><span class="err">:</span> <span class="s2">"received.domestic-corner"</span><span class="p">,</span>
  <span class="s2">"location"</span><span class="err">:</span> <span class="p">{</span>
    <span class="s2">"en"</span><span class="err">:</span> <span class="s2">"PP PRESS SHOP WAASLAND"</span><span class="p">,</span>
    <span class="s2">"fr"</span><span class="err">:</span> <span class="s2">"PP PRESS SHOP WAASLAND"</span><span class="p">,</span>
    <span class="s2">"nl"</span><span class="err">:</span> <span class="s2">"PP PRESS SHOP WAASLAND"</span><span class="p">,</span>
  <span class="p">},</span>
  <span class="s2">"time"</span><span class="err">:</span> <span class="s2">"18:04"</span><span class="p">,</span>
<span class="p">}</span>
</code></pre>
</div>

<h2 id="availableforpickup"><code class="highlighter-rouge">#availableForPickup()</code></h2>

<h3 id="description-1">Description</h3>
<p><code class="highlighter-rouge">bpost.availableForPickup()</code> will return if a package is available for pickup by the recipient.
It requires one passed parameter <code class="highlighter-rouge">id</code> and returns a true or false if the object with <code class="highlighter-rouge">knownProcessStep: 'AVAILABLE_IN_POST_POINT_INTERNATIONAL'</code> has <code class="highlighter-rouge">obj.status === 'active'</code> in the API response.</p>

<h3 id="spec-1">Spec</h3>

<table>
  <thead>
    <tr>
      <th>Parameters</th>
      <th>Return type</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code class="highlighter-rouge">(id: String)</code></td>
      <td><code class="highlighter-rouge">Promise&lt;Boolean&gt;</code></td>
    </tr>
  </tbody>
</table>

<h3 id="usage-1">Usage</h3>
<div class="language-js highlighter-rouge"><pre class="highlight"><code><span class="kr">import</span> <span class="p">{</span> <span class="nx">availableForPickup</span> <span class="p">}</span> <span class="nx">from</span> <span class="s1">'bpost'</span><span class="p">;</span>

<span class="nx">availableForPickup</span><span class="p">(</span><span class="err">`</span><span class="mi">323267009691651472</span><span class="err">`</span><span class="p">).</span><span class="nx">then</span><span class="p">(</span><span class="nx">res</span> <span class="o">=&gt;</span> <span class="nx">res</span><span class="p">);</span>
</code></pre>
</div>

<h3 id="output-1">Output</h3>
<div class="language-js highlighter-rouge"><pre class="highlight"><code><span class="kc">true</span> <span class="c1">// when processOverview.processSteps.status === 'active'</span>
<span class="kc">false</span> <span class="c1">// when processOverview.processSteps.status !== 'active'</span>
</code></pre>
</div>

</div>

        </div>
      </div>
    </div>
  </main>
</body>
</html>
