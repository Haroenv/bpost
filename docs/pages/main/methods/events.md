---
title: Getting package events
permalink: events
sidebar: main
---
## `#droppedOffBySender`

### Description
`bpost.droppedOffBySender()` will return information when the package is dropped off by the sender at sender's source postal point.
It requires one passed parameter `id` and returns the data with the `key: 'received.domestic-corner',` value in the API response.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { droppedOffBySender } from 'bpost';

droppedOffBySender(`323267009691651472`).then(res => res);
```

### Output
```js
{
  date: "2016-11-09",
  irregularity: false,
  key: "received.domestic-corner",
  location: {
    en: "PP PRESS SHOP WAASLAND",
    fr: "PP PRESS SHOP WAASLAND",
    nl: "PP PRESS SHOP WAASLAND",
  },
  time: "18:04",
}
```

## `#availableForPickup`

### Description
`bpost.availableForPickup()` will return if a package is available for pickup by the recipient.
It requires one passed parameter `id` and returns a true or false if the object with `knownProcessStep: 'AVAILABLE_IN_POST_POINT_INTERNATIONAL'` has `obj.status === 'active'` in the API response.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Boolean>`|

### Usage
```js
import { availableForPickup } from 'bpost';

availableForPickup(`323267009691651472`).then(res => res);
```

### Output
```js
true // when processOverview.processSteps.status === 'active'
false // when processOverview.processSteps.status !== 'active'
```
