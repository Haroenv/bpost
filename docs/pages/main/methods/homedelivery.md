---
title: Home Delivery
permalink: homedelivery
sidebar: main
---
## `#availableFrom`

### Description
`bpost.availableFrom()` will return the earliest timestamp that a package is availble for pickup.
It requires one passed parameter `id` returns the data with the `earliestAvailableTime: {…}` value in the API response.

{% include warning.html
  content='This method is only available if the recipient was absent on first delivery attempt.<br> It will return `undefined` if the recipient was present at first delivery attempt.'
%}

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { availableFrom } from 'bpost';

availableFrom(`323267009691651472`).then(res => res);
```

### Output
```js
{
  time: "11:00",
  day: "2016-11-23"
}
```

## `#availableTo`

### Description
`bpost.availableTo()` will return the latest timestamp that a package can be picked up.
It requires one passed parameter `id` returns the data with the `latestAvailableTime: {…}` value in the API response.

{% include warning.html
  content='This method is only available if the recipient was absent on first delivery attempt.<br> It will return `undefined` if the recipient was present at first delivery attempt.'
%}

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { availableTo } from 'bpost';

availableTo(`323267009691651472`).then(res => res);
```

### Output
```js
{
  time: "22:00",
  day: "2016-12-07"
}
```

## `#availableFromTo`

### Description
`bpost.availableFromTo()` will return a formatted string containing the period (in days) that a package is available for pickup after absence.
It requires one passed parameter `id` returns a combined data for the values of `earliestAvailableTime.day` and `latestAvailableTime.day` in the API response.

{% include warning.html
  content='This method is only available if the recipient was absent on first delivery attempt.<br> It will return `undefined` if the recipient was present at first delivery attempt.'
%}

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<String>`|

### Usage
```js
import { availableFromTo } from 'bpost';

availableFromTo(`323267009691651472`).then(res => res);
```

### Output
```js
'Available from 2016-11-23 to 2016-12-07'
```

## `#attemptedDelivery`

### Description
`bpost.attemptedDelivery()` will return the timestamp for the first delivery attempt in which recipient was absent.
It requires one passed parameter `id` returns the data with the `attemptedDeliveryTime: {…}` value in the API response.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { attemptedDelivery } from 'bpost';

attemptedDelivery(`323267009691651472`).then(res => res);
```

### Output
```js
{
  time: "12:28",
  day: "2016-11-22"
}
```

## `#actualDelivery`

### Description
`bpost.actualDelivery()` will return the timestamp for when the package was picked up at the postal point after absence.
It requires one passed parameter `id` returns the data with the `actualDeliveryTime: {…}` value in the API response.

{% include warning.html
  content='This method is only available if a package is picked up after first delivery attempt.<br> It will return `undefined` if the recipient was present at first delivery attempt or if the package is yet to be picked up at the postal point.'
%}

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { actualDelivery } from 'bpost';

actualDelivery(`323267009691651472`).then(res => res);
```

### Output
```js
{
  time: "13:37",
  day: "2016-11-23"
}
```
