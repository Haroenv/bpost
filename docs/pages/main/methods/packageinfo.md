---
title: Package Info
permalink: packageinfo
sidebar: main
---
#### packageInfo.js
__bpost#getPackageWeight()__
> Number: returns the package weight in grams

__bpost#deliveryMethod()__
> String: returns the requested delivery method
(e.g `"POSTAL_POINT"`)

__bpost#identifier()__
> Number: returns the internal id for package
_❗ this is different from the itemIdentifier param passed to the methods as first argument_

__bpost#hasRetourOptions()__
> Boolean: returns true when package is eligible for retour

__bpost#receiver(formatted: Boolean)__
> Object: returns data for the receiver
> String: returns a formatted string with receiver data
(eg. `"THIBAULT MAEKELBERGH, 8000 BRUGGE, BE"`)
