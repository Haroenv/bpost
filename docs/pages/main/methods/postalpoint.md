---
title: Postal Point
permalink: postalpoint
sidebar: main
---
## `#sourcePostalPoint`

### Description
`bpost.sourcePostalPoint()` will return the name of the postal point where the sender drops off the package.
It requires one passed parameter `id` returns the data for the `location.en` key proceeding the `"key":"received.domestic-corner"` pair.

It supports [multi-language returns](/multilanguage), either [constructed with a session](/constructing) or passed directly as a parameter.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<String>`|
|`(id: String, ?en: String)`|`Promise<String>`|

### Usage
```js
import { sourcePostalPoint } from 'bpost';

sourcePostalPoint(`323267009691651472`).then(res => res);
sourcePostalPoint(`323267009691651472`, 'nl').then(res => res);
```

### Output
```js
'PP PRESS SHOP WAASLAND'
```

## `#destinationPostalPoint`

### Description
`bpost.destinationPostalPoint()` will return the name of the postal point chosen as destination.
It requires one passed parameter `id` returns the data with the `events.deliveryPoint.name.en: …` value in the API response.

It supports [multi-language returns](/multilanguage), either [constructed with a session](/constructing) or passed directly as a parameter.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<String>`|
|`(id: String, ?lang: String)`|`Promise<String>`|

### Usage
```js
import { destinationPostalPoint } from 'bpost';

destinationPostalPoint(`323267009691651472`).then(res => res);
destinationPostalPoint(`323267009691651472`, `nl`).then(res => res);
```

### Output
```js
'POSTPUNT OCTA+ BRUGGE'
```

## `#destinationAsAddress`

### Description
`bpost.destinationAsAddress()` will return a formatted string containing the address of the destination postal point.
It requires one passed parameter `id` returns the data with the `deliveryPoint.en: …` value in the API response.

It supports [multi-language returns](/multilanguage), either [constructed with a session](/constructing) or passed directly as a parameter.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<String>`|
|`(id: String, ?lang: String)`|`Promise<String>`|

### Usage
```js
import { destinationAsAddress } from 'bpost';

destinationAsAddress(`323267009691651472`).then(res => res);
destinationAsAddress(`323267009691651472`, nl).then(res => res);
```

### Output
```js
'Pathoekeweg 74, 8000 Brugge'
```

## `#openingHours`

### Description
`bpost.openingHours()` will return the opening and closing hours the destination postal point on a given day.
It requires two passed parameters `id` and `day` and returns the data with the `deliveryPoint.openingSchedules: {…}` value in the API response.


### Spec

|Parameters|Return type|
|---|---|
|`(id: String, day: String)`|`Promise<Object>`|

### Usage
```js
import { openingHours } from 'bpost';

openingHours(`323267009691651472`, 'monday').then(res => res);
```

### Output
```js
{
  open: "07:00",
  close: "19:00",
}
```

## `#destinationCoordinate`

### Description
`bpost.destinationCoordinate()` will return a 2D lat long coordinate for the destination postal point.
It requires one passed parameter `id` returns the data with the `deliveryPoint.latitude: …` and `deliveryPoint.longitude: …` values in the API response, formatted as an object.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { destinationCoordinate } from 'bpost';

destinationCoordinate(`323267009691651472`).then(res => res);
```

### Output
```js
{
  longitude: 3.21387,
  latitude: 51.23154,
}
```
