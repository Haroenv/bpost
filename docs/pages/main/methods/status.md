---
title: Getting active status
permalink: status
sidebar: main
---
## `#getStatus`

### Description
`bpost.getStatus()` will search for the currently active step in the delivery process of the package.
It requires one passed parameter `id` and returns the data with the `status: 'active'` value in the API response.

### Spec

|Parameters|Return type|
|---|---|
|`(id: String)`|`Promise<Object>`|

### Usage
```js
import { getStatus } from 'bpost';

getStatus(`323267009691651472`).then(res => res);
```

### Output
```js
{
  knownProcessStep: "PICKED_UP_IN_POST_POINT_INTERNATIONAL",
  label: {
    detail: "inPostPoint",
    main: "pickedUp",
  },
  name: "delivered",
  status: "active",
}
```
