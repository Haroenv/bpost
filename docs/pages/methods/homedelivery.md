---
layout: method
---
#### homeDeliveryOptions.js
__bpost#availableFrom()__
> Object: returns the earliest time and day that a package can be picked up after absence

__bpost#availableTo()__
> Object: returns the latest time and day that a package can be picked up after absence

__bpost#availableFromTo()__
> String: returns a formatted string stating the period (in days) a package can be picked up after absence
(e.g `"Available from 2016-11-23 to 2016-12-07"`)

__bpost#attemptedDelivery()__
> Object: returns the time and day that delivery was attempted but receiver was absent

__bpost#actualDelivery()__
> Object: returns the time and day when a package was picked up from the postal point, receiving the absently delivered package
