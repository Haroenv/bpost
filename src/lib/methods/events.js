import find from 'lodash.find';
import composedFetch from '../composedFetch.js';

export const droppedOffBySender = id => composedFetch(id)
  .then(data => find(data.events, { key: `received.domestic-corner` }))
  .catch(e => console.error(e));

export const availableForPickup = id => {
  composedFetch(id)
  .then(data => find(
    data.processOverview.processSteps,
    { knownProcessStep: `AVAILABLE_IN_POST_POINT_INTERNATIONAL` }
  ))
  .then(obj => (obj.status === `active` ? true : false))
  .catch(e => console.error(e));
};
