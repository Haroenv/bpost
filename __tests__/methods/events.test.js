import * as bpost from '../../src/lib/methods/events';

describe(`Method: droppedOffBySender`, () => {
  it(`should return the object when package is at origin`, async () => {
    const request = await bpost.droppedOffBySender(global.BPOST_ID_POSTAL);
    expect(request).toMatchSnapshot();
  });

  it(`should throw if the id param is incorrect`, async () => {
    const request = await bpost.droppedOffBySender(`eval();`);
    expect(request).toThrow();
  });
});

describe(`Method: availableForPickup`, () => {
  it(`should return true or false according to availability`, async () => {
    const request = await bpost.availableForPickup(global.BPOST_ID_POSTAL);
    expect(request).toBeFalsy();
  });

  it(`should throw if the id param is incorrect`, async () => {
    const request = await bpost.availableForPickup(`eval();`);
    expect(request).toThrow();
  });
});
