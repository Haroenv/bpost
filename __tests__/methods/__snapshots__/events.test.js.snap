exports[`Method: droppedOffBySender should return the object when package is at origin 1`] = `
Object {
  "date": "2016-11-09",
  "irregularity": false,
  "key": "received.domestic-corner",
  "location": Object {
    "en": "PP PRESS SHOP WAASLAND",
    "fr": "PP PRESS SHOP WAASLAND",
    "nl": "PP PRESS SHOP WAASLAND",
  },
  "time": "18:04",
}
`;
